### PR TITLE
Delete Corporation

### PR DESCRIPTION
**Pull request by AdamIsrael**

### CODE REVIEW

File: actions/delete-corporation [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+set_state('vpe.delete-corporation')
+
+try:
+    main()
+except Exception as e:
+    action_fail(repr(e))
+finally:
+    remove_state('delete.add-corporation')
```
Comment (lazypower) [inline actions/delete-corporation]: incorrect state down here, it bit me too :)

File: reactive/vpe_router.py [hunk 2]
```diff
@@ -172,6 +172,105 @@ def connect_domains():
...
+        'ip',
+        'netns',
+        'exec',
+        'domain_name',
+        'ip',
+        'tun',
+        'show',
+        '|',
+        'grep',
+        'gre',
```
Comment (chuckbutler) [inline reactive/vpe_router.py]: looks wrong

File: reactive/vpe_router.py [hunk 3]
```diff
@@ -172,6 +172,105 @@ def connect_domains():
...
+    # ip netns del domain_name
+    router.ip(
+        'netns',
+        'del',
+        domain_name
+    )
+
+
+    pass
+=======
```
Comment (lazypower) [inline reactive/vpe_router.py]: merge artifact?

File: reactive/vpe_router.py [hunk 4]
```diff
@@ -196,3 +295,4 @@ def delete_domain_connection():
               'tunnel',
               'del',
               tunnel_name)
+>>>>>>> juju-solutions/master
```
Comment (lazypower) [inline reactive/vpe_router.py]: yep, merge artifact

**Comment by lazypower** [comment]
+1 LGTM now :)

File: actions/delete-corporation [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+
+set_state('vpe.delete-corporation')
+
+try:
+    main()
+except Exception as e:
+    action_fail(repr(e))
+finally:
+    remove_state('delete.delete-corporation')
```
Comment (marcoceppi) [inline actions/delete-corporation:18]: vpe.delete-corporation

File: reactive/vpe_router.py [hunk 6]
```diff
@@ -96,6 +96,101 @@ def add_corporation():
...
+            'exec',
+            domain_name,
+            'ip',
+            'tunnel',
+            'del',
+            tunnel
+        )
+
+    # Remove all interfaces associated to the domain
+    p = router._run(
```
Comment (marcoceppi) [inline reactive/vpe_router.py:150]: why not `router.ip`?

File: reactive/vpe_router.py [hunk 7]
```diff
@@ -96,6 +96,101 @@ def add_corporation():
...
+        'ip',
+        'netns',
+        'exec',
+        'domain_name',
+        'ifconfig',
+        '|',
```
Comment (marcoceppi) [inline reactive/vpe_router.py:156]: `shell != true` in subprocess, so this won't work. you'll either need to add that (don't) or do post processing in pyhton

File: reactive/vpe_router.py (hunk 6 above)
Comment (AdamIsrael) [inline reactive/vpe_router.py:150]: `router.ip` doesn't have a return, but I could modify it to do so.

**Pull request merged**